core/client.go

```diff
@@ -206,27 +194,31 @@ func (c *Client) processStream(controlStream ClientControlStream, dataStream Dat
 			if d, ok := f.(*frame.DataFrame); ok {
 				d.Clean()
 			}
-			if err != nil {
-				c.errorfn(err)
-				reconnection <- struct{}{}
-				return
-			}
+			c.handleFrameError(err, reconnection)
 		}
 	}
 }
 
-// Wait waits client error returning.
-func (c *Client) Wait() error {
-	err := <-c.shutdownChan
-	return err
-}
+func (c *Client) handleFrameError(err error, reconnection chan<- struct{}) {
```

Comment:
handleFrameError handles errors that occur during frame reading and writing by performing the following actions:
1. Sending the error to the error function (errorfn).
2. Closing the client if the data stream has been closed.
3. Always attempting to reconnect if an error is encountered.